recipes/openvdb/all/conanfile.py

```diff
@@ -27,91 +30,191 @@ class OpenVDBConan(ConanFile):
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 14)
-        if self.settings.arch not in ("x86", "x86_64"):
-            if self.options.simd:
-                raise ConanInvalidConfiguration("Only intel architectures support SSE4 or AVX.")
+        if self.settings.compiler.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
         self._check_compilier_version()
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["USE_BLOSC"] = self.options.with_blosc
+        tc.variables["USE_ZLIB"] = self.options.with_zlib
+        tc.variables["USE_LOG4CPLUS"] = self.options.with_log4cplus
+        tc.variables["OPENVDB_SIMD"] = self.options.simd
+        if Version(self.version) >= "10.0.0":
+            tc.variables["OPENVDB_USE_DELAYED_LOADING"] = self.options.delayed_load
+
+        tc.variables["OPENVDB_CORE_SHARED"] = self.options.shared
+        tc.variables["OPENVDB_CORE_STATIC"] = not self.options.shared
+
+        # All available options but not exposed yet. Set to default values
+        tc.variables["OPENVDB_BUILD_CORE"] = True
+        tc.variables["OPENVDB_BUILD_BINARIES"] = False
+        tc.variables["USE_EXR"] = False # This is only used for vdb_render
+        tc.variables["OPENVDB_BUILD_PYTHON_MODULE"] = False
+        tc.variables["OPENVDB_BUILD_UNITTESTS"] = False
+        tc.variables["OPENVDB_BUILD_DOCS"] = False
+        tc.variables["OPENVDB_BUILD_HOUDINI_PLUGIN"] = False
+        tc.variables["OPENVDB_BUILD_HOUDINI_ABITESTS"] = False
+
+        tc.variables["OPENVDB_BUILD_AX"] = False
+        tc.variables["OPENVDB_BUILD_AX_BINARIES"] = False
+        tc.variables["OPENVDB_BUILD_AX_UNITTESTS"] = False
+
+        tc.variables["OPENVDB_BUILD_MAYA_PLUGIN"] = False
+        tc.variables["OPENVDB_ENABLE_RPATH"] = False
+        tc.variables["OPENVDB_CXX_STRICT"] = False
+        tc.variables["USE_HOUDINI"] = False
+        tc.variables["USE_MAYA"] = False
+        tc.variables["USE_STATIC_DEPENDENCIES"] = False
+        tc.variables["USE_PKGCONFIG"] = False
+        tc.variables["OPENVDB_INSTALL_CMAKE_MODULES"] = False
+
+        if Version(self.version) >= "9.0.0":
+            tc.variables["OPENVDB_BUILD_NANOVDB"] = self.options.nanovdb
+            tc.variables["NANOVDB_USE_OPENVDB"] = self.options.nanovdb_with_openvdb
+            tc.variables["USE_TBB"] = self.options.nanovdb_with_tbb
+            tc.variables["NANOVDB_USE_INTRINSICS"] = True
+            tc.variables["NANOVDB_BUILD_TOOLS"] = False
+
+        if self._needs_boost:
+            tc.variables["Boost_USE_STATIC_LIBS"] = not self.options["boost"].shared
```

Comment:
```suggestion
            tc.variables["Boost_USE_STATIC_LIBS"] = not self.options["boost"].get_safe("shared", self.options.shared)
```
This fails with 
```
ConanException: option 'shared' doesn't exist
Possible options are []
```
when doing `conan create` on this recipe on a local machine. Doing `.get_safe("shared", self.options.shared)` for all `shared` option on dependencies might be a good idea.

Reply:
I've been testing your suggestion and it fails. You can't use `get_safe` on the object returned by `self.options["boost"]`.
I've also checked that the only case when `boost.options` won't contain the `shared` key is with the `header_only` version and that will fail during compilation anyways. OpenVDB, as of version 10.0.1, even with the `delayed_load` option activated requires `Boost.NumericConversion` which is not header only.
I'll leave it as it was but add a check to assure that `boost` is not in `header_only` mode.

I'm also thinking that if you could use `boost` in `header_only` mode the variable `Boost_USE_STATIC_LIBS` wouldn't make any sense.

Reply:
I was wrong. It seems `Boost.NumericConversions` is header only but there is a bug in the OpenVDB CMakeLists.txt already solved in the `master` branch.